### ISSUE TITLE
dep cycles exits 0 with cycles present; bulk import drops cycle edges

### ISSUE THREAD

**Issue by JYeswak** [Dicklesworthstone/beads_rust#368]
## What happened

br 0.2.16 (macOS arm64, release binary via `br upgrade`). Two related gaps in the same contract: **when cycle problems are detected, the process still exits 0**, so scripted/robot callers gating on exit codes can't see them.

1. `br dep cycles` finds and prints cycles — text warning and correct `--json` (`count: 1`) — but exits 0.
2. `br create -f <bulk.md>` with dependencies that would form a cycle: the transactional bulk insert correctly detects the cycle and fails, the per-edge fallback then silently drops the cycle-closing edge (stderr warning), and the command exits 0. The imported graph is missing intended edges with no machine-visible signal. (The same path's *unresolved-dependency* warnings also exit 0 — same shape.)

## Repro

```bash
d=$(mktemp -d) && cd "$d" && git init -q . && br init --prefix demo

cat > bulk.md << 'MD'
## Task A

### ID
a1

### Type
task

### Priority
2

### Description
First task.

### Dependencies
blocks: b1

## Task B

### ID
b1

### Type
task

### Priority
2

### Description
Second task.

### Dependencies
blocks: a1
MD

br create -f bulk.md; echo "BULK_RC=$?"
# warning: bulk dependency import failed (Cycle detected in dependencies: demo-xxx -> demo-yyy -> demo-xxx); retrying dependencies one by one
# warning: failed to add dependency demo-yyy → demo-xxx: Cycle detected ...
# ✓ Created 2 issues from bulk.md
# BULK_RC=0        <- exit 0; one of the two declared edges silently absent

# Seed a cycle at the storage layer to exercise the detector's exit contract.
# (br's own guards are good here: `br dep add` refuses the closing edge with a
# nonzero exit, and hand-appending the reverse edge to issues.jsonl does NOT
# survive auto-import — so a live cycle needs a direct insert to demonstrate.)
sqlite3 .beads/beads.db "INSERT INTO dependencies (issue_id, depends_on_id, type, created_at, created_by) VALUES ('<idB>','<idA>','blocks','2026-07-01T00:00:00Z','agent');"

br dep cycles; echo "RC=$?"
# Found 1 active cycle(s):
#   1. <idA> -> <idB> -> <idA>
# RC=0             <- reported, but exit 0

br dep cycles --json | jq .count   # 1
echo $?                            # 0
```

## Expected vs observed

```
br dep cycles (cycles present):     expected nonzero    observed 0
br create -f (cycle edges dropped): expected nonzero    observed 0
```

## Where it lives

- `src/cli/commands/dep.rs:1566` — `dep_cycles()` returns `Ok(())` on every path; the `count > 0` branch prints via `ctx.warning` (~:1633) and still falls through to `Ok(())`.
- `src/cli/commands/create.rs:1252-1273` — bulk-import match arm: `add_dependencies_bulk_for_import` error → `eprintln!` + per-edge retry loop whose individual failures also only `eprintln!`; no failure state escapes to the exit code.
- `src/storage/sqlite.rs:7265` — `add_dependencies_bulk_for_import` (the transactional detection itself works correctly).

## Why this matters

Bulk materialization is the agent-facing path: a swarm generates a work-graph spec, imports it with `br create -f`, and dispatches workers against `br ready`. Every step gates on exit codes. Today an import whose dependency structure was partially rejected reports success, so the swarm dispatches against a graph that differs from the approved spec — and a CI/pre-dispatch validator that runs `br dep cycles` as a gate cannot fail the gate without parsing prose output.

## Out of scope

Not asking for any change to detection or to `br dep add` — the single-edge refusal (nonzero, transactional) and auto-import's rejection of cycle edges are clearly the intended design and both behave well. This is only about the exit-code contract on the two surfaces above once detection has already fired. Implementation shape (unconditional nonzero vs. an opt-in strict flag) is your call.

**Comment by Dicklesworthstone** [comment]
Fixed in 450eb96d (+ 75916406) — `br dep cycles` and the bulk-import path now exit non-zero when cycles are detected or declared dependency edges are silently dropped, so scripted/robot callers gating on exit codes can see the condition. --json output is preserved. Thanks for the precise repro!

**Comment by JYeswak** [comment]
Confirmed fixed on `0.2.17` (built from `450eb96d` + `75916406`). Both exit-code surfaces now bite:

- `br create -f` with an unresolvable dependency → exit 5 (`DependencyNotFound`), was 0.
- `br dep cycles` on a live cycle → exit 5 (`CycleDetected`), was 0; a clean graph still exits 0.

The scripted-caller gap is closed — our pre-dispatch validator can gate on the exit code now instead of parsing the prose warning. Thanks Jeffrey.

(One note for anyone scripting against it: the signal is on the process exit code, so a wrapper that pipes the command through `| tail`/`| grep` will mask it under a bare `set -u` — needs `pipefail` or a direct `$?`. That's a caller-side gotcha, not a br bug.)
